Changed type of ignoreInVersions in visWidgets

GermanBluefox opened the pull request.

foxriver76 commented:
is it correct like this or recent change? If it was a recent design change, then we should avoid having properties with free chooseable names and fixed names on the same level.

GermanBluefox commented:
> is it correct like this or recent change? If it was a recent design change, then we should avoid having properties with free chooseable names and fixed names on the same level.
> 
> Also in the schema it is not on level of visWidgets property but on the level of the individual widgetSet.. so what is now if I want to provide one widget set exclusive for vis1 another exclusive for vis2? This would no longer be configurable..

I changed a bit

foxriver76 commented on packages/types-dev/objects.d.ts:
```diff
@@ -677,7 +675,8 @@ declare global {
             version: string;
-            visWidgets?: Record<string, VisWidget>;
+            /** The vis widget does not support the listed major versions of vis */
+            visWidgets?: Record<string, VisWidget> | { ignoreInVersions?: number[] };
```
Now you can either define via widgets or ignore I do not think it is the intention. See also my previous comment.

foxriver76 replied:
Okay, please confirm: 

Valid: 
```json
....
"visWidgets": {
    "ignoreInVersions": [2]
}
.... 
```

Valid: 
....
"visWidgets": {
    "mySuperWidget": { ... }
}
.... 
```

Invalid:
....
"visWidgets": {
    "mySuperWidget": { ... },
    "ignoreInVersions": [2]
},
.... 
```

Maybe add some type tests to `types-public` package in `index.test-d.ts` to ensure it works like intended.

foxriver76 commented:
as we have discussed in one of the core meetings the schema will stay and implementation should follow it
